cds/webapi/C#/MetadataOperations/CDSWebApiServiceExtensions.cs

```diff
@@ -0,0 +1,2458 @@
+        {
+            try
+            {
+                JObject body = new JObject()
+                {
+                    ["EntityName"] = entityName
+                };
+                return (bool)svc.Post("CanBeReferencing ", body)["CanBeReferencing"];
+            }
+            catch (ServiceException se)
+            {
+                throw new Exception($"Service Error: {se.Message}", se);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("There was a problem using the CanBeReferencing Action", ex);
+            }
+        }
+
+        /// <summary>
+        /// Checks whether an entity can participate in a many-to-many relationship.
+        /// </summary>
+        /// <param name="svc">An instance of the CDSWebApiService to extend.</param>
+        /// <param name="entitylogicalname">The logical name of the entity to check.</param>
+        /// <returns>Whether an entity can participate in a many-to-many relationship.</returns>
+        public static bool CanManyToMany(this CDSWebApiService svc, string entitylogicalname)
+        {
+            try
+            {
+                JObject body = new JObject()
+                {
+                    ["EntityName"] = entitylogicalname
+                };
+                return (bool)svc.Post("CanManyToMany", body)["CanManyToMany"];
+            }
+            catch (ServiceException se)
+            {
+                throw new Exception($"Service Error: {se.Message}", se);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("There was a problem using the CanManyToMany Action", ex);
+            }
+        }
+
+        /// <summary>
+        /// Creates an attribute and adds it to an unmanaged solution
+        /// </summary>
+        /// <param name="svc">An instance of the CDSWebApiService to extend.</param>
+        /// <param name="entityLogicalName">The logical name of the entity to add the attribute to.</param>
+        /// <param name="attribute">The attribute to add.</param>
+        /// <param name="solutionUniqueName">The unique name of the unmanaged solution to add the attribute to.</param>
+        /// <returns>The Id of the attribute and solution component</returns>
+        public static CreateAttributeResponse CreateAttribute(this CDSWebApiService svc,
+            string entityLogicalName,
+            AttributeMetadata attribute,
+            string solutionUniqueName = null)
+        {
+            Uri uri;
+            Guid attributeMetadataId;
+            Guid solutionId;
+            Guid solutionComponentId;
+
+            if (!string.IsNullOrEmpty(solutionUniqueName))
+            {
+                //Verify that solution exists and is unmanaged.
+                try
+                {
+                    solutionId = svc.CheckSolution(solutionUniqueName);
+                }
+                catch (Exception)
```

Comment:
Why catch an exception and do nothing more than throw it? Perhaps this was a placeholder for you to add some code. Anyway, this shows up in many places in your code. I would recommend removing it unless you are doing more than throwing the caught exception.

Reply:
This code is calling another method `CheckSolution ` which throws a detailed exception.
This is just verifying that the solution they named is valid before proceeding. 
Yes, it is called this way multiple times because it provides the same solution checking each time the solutionUniqueName parameter is passed.